src/containers/CourseView/index.jsx

```diff
@@ -0,0 +1,145 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import { Hyperlink, Icon, StatusAlert } from '@edx/paragon';
+
+import LabelledData from '../../components/LabelledData';
+
+const uuidRegex = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/;
+
+class CourseView extends React.Component {
+  static propTypes = {
+    match: PropTypes.shape().isRequired,
+    service: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    let error = '';
+    let loading = true;
+
+    const { uuid } = this.props.match.params;
+    if (!uuidRegex.test(uuid)) {
+      error = `Could not get course information. ${uuid} is not a valid course UUID.`;
```

Comment:
I'd like us to avoid course key if possible. I know it's a losing battle, but ideally one day, maybe we could allow changing keys.

Though I suppose allowing that doesn't mean we can't have key in URL either... URLs can change. But anyway, we can improve that later as we like.